### PR TITLE
add support with the new vec32

### PR DESCRIPTION
**Pull request by barakb**
## Release Notes

- **New Features**
  - Introduced new `Vec32` struct for handling vectors of 32-bit floating-point values.
  - Added support for new error handling variants: `ParsingF32` and `ParsingVec32`.
  - Enhanced parsing capabilities with the addition of `redis_value_as_float` and updates to the `ParserTypeMarker` enum.
  - Added a new function to parse string arrays from Redis values.
  - Enhanced index and constraint management functionalities within the graph API.

- **Documentation**
  - Updated code examples in the README to reflect changes in method calls and data handling.

- **Tests**
  - Expanded test suite with new tests for vector data handling and parsing functionalities.
  - Added tests to validate index creation, constraint management, and slowlog functionality.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,6 +24,9 @@ thiserror = "1.0.63"
 tokio = { version = "1.39.2", default-features = false, features = ["macros", "sync", "rt-multi-thread"], optional = true }
 tracing = { version = "0.1.40", default-features = false, features = ["std", "attributes"], optional = true }
 
+[dev-dependencies]
+approx = "0.5.1"
+
```
[the remark this answers is not in the document]
Comment (barakb) [reply Cargo.toml:29]: It is used on blocking but under the #[cfg(test)]

File: src/parser/mod.rs [hunk 2]
```diff
@@ -76,6 +79,11 @@ pub(crate) fn redis_value_as_double(value: redis::Value) -> FalkorResult<f64> {
         .and_then(|string_val| string_val.parse().map_err(|_| FalkorDBError::ParsingF64))
 }
 
+pub(crate) fn redis_value_as_float(value: redis::Value) -> FalkorResult<f32> {
+    redis_value_as_string(value)
+        .and_then(|string_val| string_val.parse().map_err(|_| FalkorDBError::ParsingF32))
+}
+
```
[the remark this answers is not in the document]
Comment (gkorland) [reply src/parser/mod.rs:86]: ParsingF32 added to FalkorDBError is a new enum value part of this PR

File: src/value/vec32.rs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    fn test_parse_invalid_vec32_invalid_elements() {
+        let value = redis::Value::Array(vec![
+            redis::Value::SimpleString("45.0".to_string()),
+            redis::Value::SimpleString("not a number".to_string()),
+        ]);
+        let result = Vec32::parse(value);
+        assert!(result.is_err());
+        assert_eq!(result.unwrap_err(), FalkorDBError::ParsingF32);
+    }
+}
```
[the remark this answers is not in the document]
Comment (barakb) [reply src/value/vec32.rs]: Thanks, nested array is not a legal input in this case, as for the other 2 tests I will add them

File: src/response/index.rs [hunk 4]
```diff
@@ -113,11 +132,11 @@ impl SchemaParsable for FalkorIndex {
             entity_type: parse_falkor_enum(entity_type)?,
             status: parse_falkor_enum(status)?,
             index_label: redis_value_as_typed_string(label)?,
-            fields: redis_value_as_typed_string_vec(fields)?,
             field_types: parse_types_map(field_types)?,
             language: redis_value_as_typed_string(language)?,
-            stopwords: redis_value_as_typed_string_vec(stopwords)?,
+            stopwords: parse_string_array(stopwords, graph_schema)?,
             info: parse_raw_redis_value(info, graph_schema).and_then(|val| val.into_map())?,
+            options: parse_raw_redis_value(options, graph_schema).and_then(|val| val.into_map())?,
```
[the remark this answers is not in the document]
Comment (barakb) [reply src/response/index.rs:139]: Yes this is intentionally and mentioned in a comment above the distraction, properties is intentionally ignored

File: src/response/index.rs [hunk 5]
```diff
@@ -100,7 +117,9 @@ impl SchemaParsable for FalkorIndex {
         value: redis::Value,
         graph_schema: &mut GraphSchema,
     ) -> Result<Self, FalkorDBError> {
-        let [label, fields, field_types, language, stopwords, entity_type, status, info] =
+        // properties are ignored because we deliver then in field_types and the keys of the map
+        // where the value is the index type
+        let [label, _properties, field_types, options, language, stopwords, entity_type, status, info] =
             redis_value_as_vec(value).and_then(|as_vec| {
                 as_vec.try_into().map_err(|_| {
                     FalkorDBError::ParsingArrayToStructElementCount(
```
[the remark this answers is not in the document]
Comment (barakb) [reply src/response/index.rs:125]: it is already fixed, thanks

**Review by AviAvni — APPROVED** [review]

**Pull request merged**